### PR TITLE
Add expectAsync0,1,..6 to unittest

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

**Comment by jmesserly** [comment]
I pushed another commit if you want to take a look. I didn't bother with the style cleanups (will go away in any case when we get rid of generic method comments)

**Comment by keertip** [comment]
Thanks Jen!

**Comment by keertip** [comment]
@nex3, ptal

File: lib/src/expected_function.dart [hunk 1]
```diff
@@ -102,85 +100,91 @@ class ExpectedFunction {
...
         // an error if it previously passed.
         if (_testCase.result == PASS) {
-          _testCase.error(
-              'Callback ${_id}called ($_actualCalls) after test case '
-              '${_testCase.description} had already been marked as '
-              '${_testCase.result}.$_reason');
+          _testCase
+              .error('Callback ${_id}called ($_actualCalls) after test case '
+                  '${_testCase.description} had already been marked as '
+                  '${_testCase.result}.$_reason');
```
Comment (nex3) [inline lib/src/expected_function.dart]: Please don't include unrelated formatting changes.

File: lib/src/expected_function.dart [hunk 2]
```diff
@@ -102,85 +100,91 @@ class ExpectedFunction {
...
-  _max3([a0 = _PLACEHOLDER, a1 = _PLACEHOLDER, a2 = _PLACEHOLDER]) =>
-      _max6(a0, a1, a2);
-
-  _max4([a0 = _PLACEHOLDER, a1 = _PLACEHOLDER, a2 = _PLACEHOLDER,
-      a3 = _PLACEHOLDER]) => _max6(a0, a1, a2, a3);
+  T max1([Object a0 = _PLACEHOLDER]) => max6(a0);
+
+  T max2([Object a0 = _PLACEHOLDER, Object a1 = _PLACEHOLDER]) => max6(a0, a1);
+
+  T max3([Object a0 = _PLACEHOLDER, Object a1 = _PLACEHOLDER, Object a2 = _PLACEHOLDER]) =>
```
Comment (nex3) [inline lib/src/expected_function.dart]: Long line.

File: lib/src/expected_function.dart [hunk 3]
```diff
@@ -81,9 +80,8 @@ class ExpectedFunction {
       {String id, String reason, bool isDone()})
       : this._callback = callback,
         _minExpectedCalls = minExpected,
-        _maxExpectedCalls = (maxExpected == 0 && minExpected > 0)
-            ? minExpected
-            : maxExpected,
+        _maxExpectedCalls =
+            (maxExpected == 0 && minExpected > 0) ? minExpected : maxExpected,
```
Comment (nex3) [inline lib/src/expected_function.dart]: There are still some unrelated whitespace changes...

**Comment by keertip** [comment]
Sometimes having format on save enabled is not a good idea!

**Comment by nex3** [comment]
One last thing that I really should have remembered earlier (sorry!): can you bump the pubspec to the next patch version and add a note about this to the changelog?

**Comment by keertip** [comment]
@nex3 , ptal
